tests/testutils/aci-server/server.go

```diff
@@ -380,13 +380,28 @@ func newLocalListener(port PortType, protocol ProtocolType) net.Listener {
 			portNumber = 443
 		}
 	}
-	l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", portNumber))
+	addrs, err := net.LookupHost("localhost")
 	if err != nil {
-		if l, err = net.Listen("tcp6", fmt.Sprintf("[::1]:%d", portNumber)); err != nil {
-			panic(fmt.Sprintf("aci test server: failed to listen on a port: %v", err))
+		panic(`aci test server: failed to look up "localhost", really`)
+	}
+	var lookupErrs []string
+	for try := range []int{0, 0} {
+		for _, addr := range addrs {
+			addrport := fmt.Sprintf("%s:%d", addr, portNumber)
+			l, err := net.Listen("tcp", addrport)
+			if err == nil {
+				return l
+			}
+			lookupErrs = append(lookupErrs, fmt.Sprintf("(for %s, try %d: %v)", addrport, try+1, err))
 		}
-	}
-	return l
+		// TODO: When we have discovery on a custom port then
+		// we could drop listening on fixed ports, so we
+		// probably won't get any races between old server
+		// stopping to listen and new server starting to
+		// listen.
```

Comment:
This may be superseded by abd (http://github.com/appc/abd). Will rewrite the comment.